src/Api/Controllers/EmergencyAccessController.cs

```diff
@@ -105,76 +105,76 @@ public async Task Invite([FromBody] EmergencyAccessInviteRequestModel model)
             var user = await _userService.GetUserByPrincipalAsync(User);
-            await _emergencyAccessService.InitiateAsync(new Guid(id), user);
+            await _emergencyAccessService.InitiateAsync(id, user);
         }
 
         [HttpPost("{id}/approve")]
-        public async Task Accept(string id)
+        public async Task Accept(Guid id)
         {
             var user = await _userService.GetUserByPrincipalAsync(User);
-            await _emergencyAccessService.ApproveAsync(new Guid(id), user);
+            await _emergencyAccessService.ApproveAsync(id, user);
         }
 
         [HttpPost("{id}/reject")]
-        public async Task Reject(string id)
+        public async Task Reject(Guid id)
         {
             var user = await _userService.GetUserByPrincipalAsync(User);
-            await _emergencyAccessService.RejectAsync(new Guid(id), user);
+            await _emergencyAccessService.RejectAsync(id, user);
         }
 
         [HttpPost("{id}/takeover")]
-        public async Task<EmergencyAccessTakeoverResponseModel> Takeover(string id)
+        public async Task<EmergencyAccessTakeoverResponseModel> Takeover(Guid id)
         {
             var user = await _userService.GetUserByPrincipalAsync(User);
-            var (result, grantor) = await _emergencyAccessService.TakeoverAsync(new Guid(id), user);
+            var (result, grantor) = await _emergencyAccessService.TakeoverAsync(id, user);
             return new EmergencyAccessTakeoverResponseModel(result, grantor);
         }
 
         [HttpPost("{id}/password")]
-        public async Task Password(string id, [FromBody] EmergencyAccessPasswordRequestModel model)
+        public async Task Password(Guid id, [FromBody] EmergencyAccessPasswordRequestModel model)
         {
             var user = await _userService.GetUserByPrincipalAsync(User);
-            await _emergencyAccessService.PasswordAsync(new Guid(id), user, model.NewMasterPasswordHash, model.Key);
+            await _emergencyAccessService.PasswordAsync(id, user, model.NewMasterPasswordHash, model.Key);
         }
 
         [HttpPost("{id}/view")]
-        public async Task<EmergencyAccessViewResponseModel> ViewCiphers(string id)
+        public async Task<EmergencyAccessViewResponseModel> ViewCiphers(Guid id)
         {
             var user = await _userService.GetUserByPrincipalAsync(User);
-            var viewResult = await _emergencyAccessService.ViewAsync(new Guid(id), user);
+            var viewResult = await _emergencyAccessService.ViewAsync(id, user);
             return new EmergencyAccessViewResponseModel(_globalSettings, viewResult.EmergencyAccess, viewResult.Ciphers);
         }
 
         [HttpGet("{id}/{cipherId}/attachment/{attachmentId}")]
-        public async Task<AttachmentResponseModel> GetAttachmentData(string id, string cipherId, string attachmentId)
+        public async Task<AttachmentResponseModel> GetAttachmentData(Guid id, string cipherId, string attachmentId)
```

Comment:
I believe we could change `cipherId` to also be a `Guid`, if we update the `EmergencyAccessService.GetAttachmentDownloadAsync` to use a guid directly instead of casting it later.

Reply:
Done. Sorry for the delay. See f39d9b51e1494c98d2e2692e5dd7e86d72ec3594

Reply:
No problems, thanks for making the change!